packages/compiler-core/src/transforms/transformElement.ts

```diff
@@ -909,15 +913,51 @@ export function buildDirectiveArgs(
       }
       dirArgs.push(`void 0`)
     }
+
     const trueExpression = createSimpleExpression(`true`, false, loc)
-    dirArgs.push(
-      createObjectExpression(
-        dir.modifiers.map(modifier =>
-          createObjectProperty(modifier, trueExpression),
+
+    const staticMods: ExpressionNode[] = []
+    const callArgs: (ObjectExpression | ExpressionNode)[] = []
+
+    for (let i = 0; i < dir.modifiers.length; i++) {
+      const modifier = dir.modifiers[i]
+
+      if ((modifier as SimpleExpressionNode).isStatic) {
+        staticMods.push(modifier)
+      } else {
+        // Object expression must be first to avoid null or undefined
+        if (staticMods.length || i === 0) {
```

Comment:
L930-L937 is the same as L946-L953. Maybe it should be extracted as a method for reusing.